scripts/doc_checker.py

```diff
@@ -358,14 +365,15 @@ def numpydoc_validate(path: pathlib.Path) -> bool:
     for root, _, files in walker:
         if "__pycache__" in root:
             continue
-        for f in files:
-            if not f.endswith(".py"):
+        for _file in files:
+            if not _file.endswith(".py"):
                 continue
 
-            current_path = os.path.join(root, f)
+            current_path = os.path.join(root, _file)
             # get importable name
             module_name = current_path.replace("/", ".").replace("\\", ".")
-            module_name = module_name.replace(".py", "")
+            # remove ".py"
+            module_name = module_name[:-3]
```

Comment:
```suggestion
            module_name = os.path.splitext(module_name)[0]
```
[follow the manual, Luke](https://docs.python.org/3/library/os.path.html#os.path.splitext)